File: CPP/Program/DetailBarang.cpp
#include <iostream>
// include file barang
#include "Barang.cpp"

using namespace std;

class DetailBarang : public Barang { // inheritance dari class Barang
    // atribut private
    private:
        string nama_barang;
        double harga;
        int stok;

    public:
        // constructor kosong
        DetailBarang(){}

        // constructor
        DetailBarang(string id_kategori, string nama_kategori, string deskripsi, 
            string merk, string model, int tahun_produksi, 
            string nama_barang, double harga, int stok) : Barang (id_kategori, nama_kategori, deskripsi, merk, model, tahun_produksi){ // manggil constructor parent
            this->nama_barang = nama_barang;
            this->harga = harga;
            this->stok = stok;
        }

        // getter
        string getNamaBarang() const {
            return nama_barang;
        }
        double getHarga() const {
            return harga;
        }
        int getStok() const {
            return stok;
        }

        // setter
        void setNamaBarang(string nama_barang) {
            this->nama_barang = nama_barang;
        }
        void setHarga(double harga) {
            this->harga = harga;
        }
        void setStok(int stok) {
            this->stok = stok;
        }

        // destructor
        ~DetailBarang() { }
};

File: Python/Program/DetailBarang.py
# include Barang.py
from Barang import Barang

class DetailBarang(Barang):
    def __init__(self, id_kategori="", nama_kategori="", deskripsi="",
                 merk="", model="", tahun_produksi=0,
                 nama_barang="", harga=0.0, stok=0):
        super().__init__(id_kategori, nama_kategori, deskripsi, merk, model, tahun_produksi)
        self.__nama_barang = nama_barang
        self.__harga = harga
        self.__stok = stok

    # getter
    def get_nama_barang(self):
        return self.__nama_barang
    def get_harga(self):
        return self.__harga
    def get_stok(self):
        return self.__stok

    # setter
    def set_nama_barang(self, nama_barang):
        self.__nama_barang = nama_barang
    def set_harga(self, harga):
        self.__harga = harga
    def set_stok(self, stok):
        self.__stok = stok
